openssh/USAGE.md

```diff
@@ -0,0 +1,270 @@
+- `KexAlgorithms`: Comma-separated list of enabled key-exchange algorithms. Priority given by order. Names according to [this KEX naming scheme](https://github.com/open-quantum-safe/openssh#key-exchange).
+- `HostKey`: Path to host key files. One entry for one file, can have multiple entries. SSHD will look for those files when offering a host key to a client. Names of the key files need to be `/opt/oqs-ssh/ssh_host_<SIG>_key` in order for them to be successfully generated. `<SIG>` is a post-quantum signature algorithm according to [this SIG naming scheme](https://github.com/open-quantum-safe/openssh#digital-signature), with every `-` replace by `_`. A list with of some possible `HostKey` values can be found [here](https://github.com/open-quantum-safe/oqs-demos/tree/main/openssh/sshd_config).
+- `HostKeyAlgorithms`: Comma-separated list of the host offers to the client. Priority given by order. Note that if no corresponding `HostKey` is specified, this algorithm is ignored. Names according to [this SIG naming scheme](https://github.com/open-quantum-safe/openssh#digital-signature)
+- `PubkeyAcceptedKeyTypes`: Comma-separated list of public keys the host will accept from the client.
+
+Be aware that configurations for `sshd` and `ssh` may be entirely different as long as you don't want to connect to `localhost`. For example can `ssh`  be configured like a classical `ssh` client and `sshd` may support post-quantum algorithms only with no backwards compatibility at all.
+
+*Note(s):*
+- After changing any of the settings above, don't forget to `rc-service oqs-sshd restart` to apply the changes.
+- After adding either a new host or identity key, run `/opt/oqs-ssh/key-gen.sh` to generate the key files as specified. Already existing files won't be altered during this process.
+- New keys won't have any effect if they are not configured to be used in `HostKeyAlgorithms` or `PubkeyAcceptedKeyTypes` or if other keys have higher priority (first listed in `HostKeyAlgorithms`/`PubkeyAcceptedKeyTypes` has highest priority).
+
+## Automatic key generation
+
+The generation of the host and identity keys happens via the script [key-gen.sh](https://github.com/open-quantum-safe/oqs-demos/tree/main/openssh/key-gen.sh) that is executed automatically every time the container started or run. The script checks if the required key already exist and gerenates it if necessary. This script is called every time the container is started, meaning the first time the container is run (`docker run ...`) and it is startet again (`docker start ...`) after a reboot or a `docker stop ...` command. It checks for existing keys before it generates new ones so it will **never** overwrite an already existing key. It also makes sure `sshd` is startet after at boot time or is restartet after new host keys were generated.
+
+Which keys to generate is determined using the configuration files (`ssh_config` and `sshd_config`). A key's necessity is determinded based on the following parameters:
```

Comment:
Nit: "determinded" -> "determined" or maybe even "The need for a specific key is determined based..."